src/models.ts

```diff
@@ -44,6 +166,228 @@ export class FlagConfiguration {
+}
+
+export class Links {
+	self?: Link;
+	next?: Link;
+}
+
+export class Id {
+	static discriminator: string | undefined = undefined;
+
+	static attributeTypeMap: Array<{ name: string; baseName: string; type: string }> = [];
+
+	static getAttributeTypeMap() {
+		return Id.attributeTypeMap;
+	}
+}
+
+export class Role {
+	static discriminator: string | undefined = undefined;
+
+	static attributeTypeMap: Array<{ name: string; baseName: string; type: string }> = [];
+
+	static getAttributeTypeMap() {
+		return Role.attributeTypeMap;
+	}
+}
+
+export class Member {
+	links?: Links;
+	id?: Id;
+	role?: Role;
+	email?: string;
+	firstName?: string;
+	lastName?: string;
+	pendingInvite?: boolean;
+	isBeta?: boolean;
+	customRoles?: Array<Id>;
+}
+
+export class Defaults {
+	/**
+	 * The index of the variation to be served when a flags targeting is on (default variation).
+	 */
+	onVariation: number;
+	/**
+	 * The index of the variation to be served when a flag is off.
+	 */
+	offVariation: number;
+
+	static discriminator: string | undefined = undefined;
+
+	static attributeTypeMap: Array<{ name: string; baseName: string; type: string }> = [
+		{
+			name: 'onVariation',
+			baseName: 'onVariation',
+			type: 'number',
+		},
+		{
+			name: 'offVariation',
+			baseName: 'offVariation',
+			type: 'number',
+		},
+	];
+
+	static getAttributeTypeMap() {
+		return Defaults.attributeTypeMap;
+	}
+}
+
+export class FeatureFlag {
+	constructor(init?: Partial<FeatureFlag>) {
+		Object.assign(this, init);
+	}
+
+	key?: string;
+	/**
+	 * Name of the feature flag.
+	 */
+	name?: string;
+	/**
+	 * Description of the feature flag.
+	 */
+	description?: string;
+	/**
+	 * Whether the feature flag is a boolean flag or multivariate.
+	 */
+	kind?: string;
+	/**
+	 * A unix epoch time in milliseconds specifying the creation time of this flag.
+	 */
+	creationDate?: number;
+	includeInSnippet?: boolean;
+	/**
+	 * Whether or not this flag is temporary.
+	 */
+	temporary?: boolean;
+	/**
+	 * The ID of the member that should maintain this flag.
+	 */
+	maintainerId?: string;
+	/**
+	 * An array of tags for this feature flag.
+	 */
+	tags?: Array<string>;
+	/**
+	 * The variations for this feature flag.
+	 */
+	variations?: Array<Variation>;
+	/**
+	 * An array goals from all environments associated with this feature flag
+	 */
+	goalIds?: Array<string>;
+	version?: number;
+	/**
+	 * A mapping of keys to CustomProperty entries.
+	 */
+	customProperties?: { [key: string]: CustomProperty };
+	links?: Links;
+	maintainer?: Member;
+	environments?: { [key: string]: FeatureFlagConfig };
+	/**
+	 * A unix epoch time in milliseconds specifying the archived time of this flag.
+	 */
+	archivedDate?: number;
+	/**
+	 * Whether or not this flag is archived.
+	 */
+	archived?: boolean;
+	defaults?: Defaults;
+}
+
+export class PatchOperation {
+	op: string;
+	path: string;
+	value: any;
+}
+export class PatchComment {
+	comment?: string;
+	patch?: Array<PatchOperation>;
+}
```

Comment:
per the last comment on patching - these types might not be needed anymore. Let's limit the models we define manually to what we actually need until we're able to use the generated openapi client